File: backend/app/models/inventory.py
from enum import Enum


class MovementType(str, Enum):
    RECEIPT = "receipt"
    PICK = "pick"
    PACK = "pack"
    DISPATCH = "dispatch"
    ADJUSTMENT = "adjustment"
    TRANSFER = "transfer"
    COUNT_ADJUSTMENT = "count_adjustment"
    # Ajuste que deja el WMS igual a Defontana. No viaja al ERP: ya tiene esas cantidades.
    RECONCILIATION = "reconciliation"
    # Corrección de la IDENTIDAD del lote de un saldo (lote mal ingresado -> el correcto), sin
    # cambiar la cantidad. No viaja al ERP: Defontana ya manda las cantidades y los lotes.
    LOT_CORRECTION = "lot_correction"


class ReferenceType(str, Enum):
    ORDER = "order"
    PICKING_TASK = "picking_task"
    PACKING_TASK = "packing_task"
    DISPATCH = "dispatch"  # salida por despacho (permite revertir por guía)
    INVENTORY_COUNT = "inventory_count"
    MANUAL = "manual"
    RECONCILIATION = "reconciliation"  # conciliación WMS ← Defontana

File: frontend/src/lib/inventory.ts
/**
 * Presentación de los datos de inventario.
 *
 * Los valores internos no cambian: son los de `MovementType` en el backend.
 */
const MOVEMENT_LABELS: Record<string, string> = {
  receipt: 'Recepción',
  pick: 'Picking',
  pack: 'Packing',
  dispatch: 'Despacho',
  adjustment: 'Ajuste',
  transfer: 'Transferencia',
  count_adjustment: 'Ajuste por conteo',
  // Deja el WMS igual a Defontana; no viaja al ERP.
  reconciliation: 'Conciliación con ERP',
};

/** Tipo de movimiento en español; si es desconocido, se muestra el valor original. */
export function movementLabel(type?: string | null): string {
  if (!type) return '—';
  return MOVEMENT_LABELS[type] ?? type;
}
